src/coreclr/src/jit/importer.cpp

```diff
@@ -8259,7 +8243,8 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
     //-------------------------------------------------------------------------
     // The "this" pointer
 
-    if (!(mflags & CORINFO_FLG_STATIC) && !((opcode == CEE_NEWOBJ) && (newobjThis == nullptr)))
+    if (((mflags & CORINFO_FLG_STATIC) == 0) && ((sig->callConv & CORINFO_CALLCONV_EXPLICITTHIS) == 0) &&
```

Comment:
This looks like an independent bugfix. Should it be submitted in separate PR, with the regression test?

Reply:
Sure, I'll extract this fix and the regression test I added into a separate PR.

Reply:
#35385 has the fix and a regression test for the explicit this bug (#35384). Once it's in, I will merge master again to this branch.

Reply:
Pushed a merge to this branch.